docs: add a functional feature checklist for ORAS SDKs (#515)
oras-project/oras-www: versioned_docs/version-1.3/client_libraries/overview.mdx

```go
func ExtendedCopy(from Target, fromRef string, to Target, toRef string) Descriptor
```

## SDK Features

> This section defines a shared goals and roadmap framework for all ORAS SDKs, grouping features into Core, Recommended, and Optional to align implementation priorities and user expectations.

Reviewer: ```suggestion
This section defines the shared goals and roadmap for all ORAS SDKs, grouping features into Core, Recommended, and Optional to align implementation priorities and user expectations.
```
